XiaomiSmartHome/XiaomiSmartHome/Program.cs

```diff
@@ -0,0 +1,236 @@
+                    if (PackageHost.TryGetSettingValue<bool>("ReportLog", out ReportLog))
+                    {
+                        if (ReportLog)
+                        {
+                            PackageHost.WriteInfo("{0}", response);
+                        }
+                    }
+                    else
+                    {
+                        PackageHost.WriteError("Impossible de récupérer le setting 'ReportLog' en boolean");
+                    }
+
+                    //// If model equal gateway
+                    if ((data.model == "gateway") && (data.sid == Gateway.Sid))
+                    {
+
+                        //// We deserialize report data as gateway class
+                        gateway_report datagate = JsonConvert.DeserializeObject<gateway_report>(data.data);
+                        Gateway.Report = datagate;
+                        
+                        //// We push the gateway SO
+                        PackageHost.PushStateObject<gateway>("Gateway", Gateway);
+                    }
+                    else
+                    {
+
+                        //// If equipement sid is in the equipements list
+                        if (equipements.Contains(data.sid))
+                        {
+                            SaveEquipement(data.sid);
+                        }
+                        
+                    }
+
+                }
+                Thread.Yield();
+            }
+        }
+        
+        public void GetEquipements()
+        {
+
+            if (PackageHost.TryGetSettingValue<string>("GatewayIP", out GatewayIP))
+            {
+                //// Get equipements list
+                EquipementsList list = EquipementsList.FindEquipementsList(GatewayIP);
+
+                //// Get data part
+                equipements = EquipementsList.GetListData(list.data);
+
+                //// Foreach equipement in data
+                foreach (string equipement in equipements)
+                {
+                    SaveEquipement(equipement);
+                }
+            }
+            else
+            {
+                PackageHost.WriteError("Impossible de récupérer le setting 'GatewayIP' en string");
+            }
+        }
+
+        public void SaveEquipement(string mac)
+        {
+            //// Try to get setting based on equipement SID
+            if (!PackageHost.TryGetSettingValue<string>(mac, out name))
+            {
+                name = mac;
+            }
+
+            //// Try to get gateway IP adress
+            if (PackageHost.TryGetSettingValue<string>("GatewayIP", out GatewayIP))
+            {
+
+                //// Get response from read function
+                Response read = Response.ReadEquipement(GatewayIP, mac);
+
+                //// Create instance of model class
+                dynamic model  = System.Reflection.Assembly.GetExecutingAssembly().CreateInstance("XiaomiSmartHome.Equipement." + read.model);
```

Comment:
J'ai essayé avec la gateway, j'ai utilisé 
`                            Type equipmentType = Assembly.GetExecutingAssembly().GetTypes().SingleOrDefault(t => t.GetCustomAttribute<XiaomiEquipementAttribute>().Model == "gateway");
                            var test = equipmentType;
                            PackageHost.WriteInfo("{0}", equipmentType);`

Mais cela me provoque le crash du package.

Dans gateway.cs j'ai mis 

`    [StateObject, XiaomiEquipement("gateway")]
    public class Gateway`

et

`    public class XiaomiEquipementAttribute : Attribute
    {
        public string Model { get; set; }

        public XiaomiEquipementAttribute(string model)
        {
            this.Model = model;
        }
    }`